packages/cursorless-vscode/src/ide/vscode/VscodeTextLineImpl.ts

```diff
@@ -21,12 +21,16 @@ export default class VscodeTextLineImpl implements TextLine {
     return fromVscodeRange(this.line.rangeIncludingLineBreak);
   }
 
-  get firstNonWhitespaceCharacterIndex(): number {
-    return this.line.firstNonWhitespaceCharacterIndex;
-  }
-
-  get lastNonWhitespaceCharacterIndex(): number {
-    return this.line.text.trimEnd().length;
+  get rangeTrimmed(): Range {
+    return this.line.isEmptyOrWhitespace
```

Comment:
Something feels awkward about all these extra is empty checks. I think maybe we need to think a bit more carefully about this one. Like maybe we should return undefined or something special when it's empty or whitespace